lib/pages/chat/chat.dart

```diff
@@ -1776,39 +1765,48 @@ class ChatController extends State<Chat>
     }
   }
 
-  List<PopupMenuEntry<ChatAppBarActions>> appBarActionsBuilder() {
-    final listAction = [
-      if (PlatformInfos.isAndroid) ...[
-        if (selectedEvents.length == 1 &&
-            selectedEvents.first.hasAttachment &&
-            !selectedEvents.first.isVideoOrImage)
-          ChatAppBarActions.saveToDownload,
-      ],
-      if (selectedEvents.length == 1 && selectedEvents.first.isVideoOrImage)
-        ChatAppBarActions.saveToGallery,
-      ChatAppBarActions.info,
-      ChatAppBarActions.report,
-    ];
-    return listAction
-        .map(
-          (action) => PopupMenuItem(
-            value: action,
-            child: Row(
-              mainAxisSize: MainAxisSize.min,
-              children: [
-                Icon(
-                  action.getIcon(),
-                  color: action.getColorIcon(context),
-                ),
-                Padding(
-                  padding: action.getPaddingTitle(),
-                  child: Text(action.getTitle(context)),
-                ),
-              ],
-            ),
-          ),
-        )
-        .toList();
+  void handleAppbarMenuAction(
+    BuildContext context,
+    TapDownDetails tapDownDetails,
+  ) {
+    final offset = tapDownDetails.globalPosition;
+    showTwakeContextMenu(
+      offset: offset,
+      context: context,
+      builder: (_) => _appbarMenuActionTile(context),
+    );
+  }
+
+  List<Widget> _appbarMenuActionTile(BuildContext context) {
```

Comment:
please separate app bar for: select mode and normal mode

Reply:
done